core/shared/src/main/scala/cats/parse/Rfc5234.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2020 Typelevel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package cats.parse
+
+/** Parsers for the common rules of RFC5234. These rules are
+  * referenced by several RFCs.
+  *
+  * @see [[https://tools.ietf.org/html/rfc5234]]
+  */
+object Rfc5234 {
+
+  /** A-Z and a-z, without diacritics
+    */
+  val alpha: Parser1[Char] =
+    Parser.charIn('A' to 'Z').orElse1(Parser.charIn('a' to 'z'))
+
+  /** `0` or `1`
+    */
+  val bit: Parser1[Char] =
+    Parser.charIn('0' to '1')
+
+  /** any 7-bit US-ASCII character, excluding NUL
+    */
+  val char: Parser1[Char] =
+    Parser.charIn(0x01.toChar to 0x7f.toChar)
+
+  /** carriage return
+    */
+  val cr: Parser1[Unit] =
+    Parser.char('\r')
+
+  /** linefeed
+    */
+  val lf: Parser1[Unit] =
+    Parser.char('\n')
+
+  /** Internet standard newline */
+  val crlf: Parser1[Unit] =
+    Parser.string1("\r\n")
+
+  /** controls */
+  val ctl: Parser1[Char] =
+    Parser.charIn(0x7f, (0x00.toChar to 0x1f.toChar: _*))
+
+  /** `0` to `9`
+    */
+  val digit: Parser1[Char] =
+    Parser.charIn('0' to '9')
```

Comment:
Outside of this PR; but I was just wondering if we could explore the idea of reusable `Parser` i.e this one is already available under `Numbers#digit`.

I see some other defined here that could be useful for other use cases.

Reply:
I was trying to hew closely to the spec, but the test now does so.  We could use the `Numbers` version.